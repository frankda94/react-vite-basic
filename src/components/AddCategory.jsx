import { useState } from "react"
import PropTypes from 'prop-types';


export const AddCategory = ({ onNewCategory }) => {


    const [inputValue, setInputValue] = useState('');

    const onInputChage = ({ target }) => {
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;
        // setCategories(categories => [...categories, inputValue])
        onNewCategory(inputValue);
        setInputValue('');
        // console.log(setCategories);

    }

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={onInputChage}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}