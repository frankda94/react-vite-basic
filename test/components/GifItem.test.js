import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('test gifItem', () => {

    const tittle = 'tituloImagen';
    const url = 'https://urlimagen/';

    test('should to do match with snapshot', () => {
        const { container } = render(<GifItem title={tittle} url={url} />);
        expect(container).toMatchSnapshot();
    })

    test('should show image with URL and Alt indicated', () => {
        render(<GifItem title={tittle} url={url} />);
        expect(screen.getByRole('img').src).toBe(url);
        expect(screen.getByRole('img').alt).toBe(tittle);
    })
    

})
