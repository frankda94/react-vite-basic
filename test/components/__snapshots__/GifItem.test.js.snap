// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test gifItem should to do match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="tituloImagen"
      src="https://urlimagen/"
    />
  </div>
</div>
`;
